src/types/GracefulError.ts

```diff
@@ -0,0 +1,12 @@
+export interface GracefulError {
+  message: string;
+  type?: string;
+  stackTrace?: string;
+}
+
+export const isOfTypeGracefulError = (tbd: any): tbd is GracefulError => {
+  if ((tbd as GracefulError).message) {
+    return true;
+  }
+  return false;
+};
```

Comment:
I wonder if this type guard can be more specific. Technically this will return true if any object with a `message` property is passed in. E.g. if I pass in `{ message: 'hello', disclaimer: 'this is not a graceful error but tricks the type guard' }`, it will return true.

I don't think this is a huge deal right now. I'm trying to think of a way to update it that doesn't require too much code, but it would probably require some piece of identification on the interface that we could check. Unfortunately that means everywhere we make a graceful error, we'd have to add a new property. That seems like too much work for now, so I'm fine leaving this. Just worth noting for the future